apps/explorer/tests/transaction.spec.ts

```diff
@@ -12,3 +17,47 @@ test('displays gas breakdown', async ({ page }) => {
     await page.goto(`/txblock/${txid}`);
     await expect(page.getByTestId('gas-breakdown')).toBeVisible();
 });
+
+test('displays inputs', async ({ page }) => {
+    const address = await faucet();
+    const tx = await split_coin(address);
+    const txid = getTransactionDigest(tx);
+    await page.goto(`/txblock/${txid}`);
+
+    const transactionKindName = getTransactionKindName(getTransactionKind(tx)!);
+
+    if (transactionKindName === 'ProgrammableTransaction') {
+        await expect(page.getByTestId('inputs-card')).toBeVisible();
+
+        const programmableTxn = tx.transaction!.data
+            .transaction as ProgrammableTransaction;
+        const actualInputsCount = programmableTxn.inputs.length;
+
+        const inputsCardContentsCount = await page
+            .getByTestId(`inputs-card-content`)
+            .count();
+        await expect(inputsCardContentsCount).toBe(actualInputsCount);
+    }
+});
+
+test('displays transactions card', async ({ page }) => {
+    const address = await faucet();
+    const tx = await split_coin(address);
+    const txid = getTransactionDigest(tx);
+    await page.goto(`/txblock/${txid}`);
+
+    const transactionKindName = getTransactionKindName(getTransactionKind(tx)!);
+
+    if (transactionKindName === 'ProgrammableTransaction') {
```

Comment:
We should just make sure we're testing a programmable transaction page always

Reply:
Put another way, you shouldn't need this guard.

Reply:
@Jordan-Mysten updated. I am assuming `split_coin` will always result in a prog txn so I removed the check. Let me know if otherwise.

Reply:
Yeah that's a good assumption, and if it doesn't it should break the tests.